router/image/templates/ssl.conf

```diff
@@ -0,0 +1,23 @@
+# set the allowed protocols
+ssl_protocols {{ or (getv "/deis/router/sslProtocols") "TLSv1 TLSv1.1 TLSv1.2" }};
+
+# turn on session caching to drastically improve performance
+ssl_session_cache {{ or (getv "/deis/router/sslSessionCache") "shared:SSL:32m" }};
```

Comment:
@bacongobbler you are right. This is how cloudflare handle the sharing part https://blog.cloudflare.com/tls-session-resumption-full-speed-and-secure/ and this is what could provide this to nginx using lua https://github.com/openresty/lua-nginx-module/issues/331 (but is not ready, yet)